service/application/src/main/java/com/solace/maas/ep/event/management/agent/command/SempPatchCommandManager.java

```diff
@@ -0,0 +1,124 @@
+    }
+
+    public void execute(Command command, SempApiProvider sempApiProvider) {
+        try {
+            validate(command, sempApiProvider);
+            executeSempPatchCommand(command, sempApiProvider);
+            command.setResult(CommandResult.builder()
+                    .status(JobStatus.success)
+                    .logs(List.of())
+                    .build());
+        } catch (Exception e) {
+            log.error("SEMP patch command not executed successfully", e);
+            // SEMP APIExceptions don't expose the response body via e.getMessage()
+            setCommandError(command, e);
+        }
+    }
+
+    private void executeSempPatchCommand(Command command, SempApiProvider sempApiProvider) throws ApiException, JsonProcessingException {
+        String entityType = (String) command.getParameters().get(SEMP_PATCH_ENTITY_TYPE);
+        SempEntityType patchEntityType;
+        if (entityType == null) {
+            throw new IllegalArgumentException("Entity type of a SEMP patch command must not be null");
+        }
+        try {
+            patchEntityType = SempEntityType.valueOf(entityType);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Unsupported SEMP patch entity type:" + entityType);
+        }
+        switch (patchEntityType) {
+            case solaceRdpRestConsumer:
+                executePatchRdpRestConsumer(command, sempApiProvider);
+                break;
+            default:
+                throw new UnsupportedOperationException("Unsupported entity type for patch: " + patchEntityType);
+        }
+    }
+
+    // Suppressing PMD of unused entityName
+    // This is because we are not logging the entityName for security reasons now
+    // We might adapt that in the future to log the entityName based on logging level and properties file
+    @SuppressWarnings("PMD")
+    private void handleSempApiPatchException(ApiException e, String entityType) throws ApiException {
+        // If the entity does not exist, we don't want to consider it an error
+        // This is because of the edge case of dangling entities from previous deployment that did not get finalized
+        // SEMP does not report a 404 for entities that do not exist, so we have to check the response body
+        // NOT_FOUND is the string that SEMP returns within the response JSON, when an entity does not exist
+        if (e.getCode() == 400 && e.getResponseBody().contains("NOT_FOUND")) {
+            // we don't want to log the content of the response body, as it may contain sensitive information
+            // we only log the entity type. The entity name is also considered sensitive information and is not logged
+            log.info("SEMP delete: tried to patch {} which did not exist (anymore)", entityType);
+        } else {
+            throw e;
+        }
+    }
+
+
+    private static void validate(Command command, SempApiProvider sempApiProvider) {
```

Comment:
why `static` ?

Reply:
Looks like it doesn't have anything that requires an instance variable and it's like a utility method, so if it can be static, it should.